test/language/class/class-strict-mode-with.js

```diff
@@ -0,0 +1,15 @@
+// Copyright (C) Copyright 2014 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 14.5
+description: >
+    class strict mode, with
+flags: [noStrict]
+negative: SyntaxError
+---*/
+with ({a: 1}) {
+  assert.sameValue(a, 1);
+}
+
+class C extends (function B() { with ({}); return B; }()) {}
```

Comment:
The outer `with` can be removed, making this viable in either mode. `class C extends (function B() { with ({}); return B; }()) {}` is a syntax error in all modes